cilium/cmd/kvstore.go

```diff
@@ -54,6 +56,15 @@ func setupKvstore() {
 				kvStoreOpts[k] = v
 			}
 		}
+
+		k8sclient.Configure("", "/var/lib/cilium/cilium.kubeconfig")
```

Comment:
This is a leftover, it needs to be fixed as when using `cilium kvstore get` the cilium CLI command does not have the necessary configuration to contact kubernetes, to retrieve the certificates to contact etcd.

Reply:
A few questions: 
* Is the kubeconfig always guaranteed to be at this path? I presume yes because we are mounting it as such in the Cilium DS, but this is user-configurable, yes?
* This will always result in `k8sclient.IsEnabled()` to return true:

https://github.com/cilium/cilium/blob/db674150168541c2d1dc6524d761f7e2d0f00906/pkg/k8s/config.go#L60-L66

`IsEnabled` returns true if the KubeconfigPath is non-nil, and you unconditionally set KubeconfigPath above in your call to `k8sclient.Configure`:

https://github.com/cilium/cilium/blob/db674150168541c2d1dc6524d761f7e2d0f00906/pkg/k8s/config.go#L48-L58

Won't this break non-K8s setups?

Reply:
I think the path is returned by `/config` in https://github.com/cilium/cilium/blob/master/api/v1/models/daemon_configuration_status.go#L33:2 (I noticed this being printed in https://github.com/cilium/cilium/blob/master/cilium/cmd/config.go#L80:59

That might simplify things a bit?